src/Http/Routing/test/UnitTests/LinkGeneratorIntegrationTest.cs

```diff
@@ -121,10 +121,20 @@ public LinkGeneratorIntegrationTest()
                         "Admin/{controller=Home}/{action=Index}/{id?}",
                         defaults: new { area = "Admin", },
                         parameterPolicies: new { area = "Admin", },
-                        requiredValues: new { controller = RoutePattern.RequiredValueAny, action = RoutePattern.RequiredValueAny, area = "Admin", page = (string)null,}),
+                        requiredValues: new { controller = RoutePattern.RequiredValueAny, action = RoutePattern.RequiredValueAny, area = "Admin", page = (string)null, }),
                     order: 1000,
                     metadata: new object[] { new SuppressMatchingMetadata(), }),
 
+                // Conventional routed link generation route 3 - this doesn't match any actions.
+                EndpointFactory.CreateRouteEndpoint(
+                    RoutePatternFactory.Parse(
+                        "api/{controller}/{id?}",
+                        defaults: new { },
+                        parameterPolicies: new { },
+                        requiredValues: new { controller = RoutePattern.RequiredValueAny, action = (string)null, area = (string)null, page = (string)null, }),
```

Comment:
What will happen when someone adds a route like this: `routes.MapRoute("foo", "/foo/foo");`

Are there are no required values, and that means no RequiredValueAny?

Reply:
It would have required value null for everything that MVC thinks is special.

Reply:
Maybe add an integration test for it? 🙏 

(or create an issue for it if you don't want to do it now)













(or tell me to piss off 😄 )

Reply:
Added a test